common/engine/Match.lua

```diff
@@ -12,6 +12,7 @@ local LegacyPanelSource = require("common.compatibility.LegacyPanelSource")
 local InputCompression = require("common.data.InputCompression")
 local ReplayV3 = require("common.data.ReplayV3")
 local MatchRules = require("common.data.MatchRules")
+local DebugSettings = require("client.src.debug.DebugSettings")
```

Comment:
We should not introduce a dependency to the client here as it means that we would have to deploy client code on the server which does not make sense.
I think ideally `Match` should have its own internal debug configuration that is always set on construction and defaults to non-debug values, e.g.
```Lua
Match.debug = {
  enabled = false,
  vsFramesBehind = 0
}
```
The client can then update the match configuration from its client configuration after it was created and `Match` only needs to check values internally that are guaranteed to exist.